### PR TITLE
Synchronizes JSON endpoints code as of 19th of April 2017

### PR DESCRIPTION
**Pull request by zinigor**
Syncs JSON endpoints code with dotcom endpoints code.

### CODE REVIEW

File: json-endpoints/class.wpcom-json-api-list-media-v1-1-endpoint.php [hunk 1]
```diff
@@ -97,6 +97,20 @@ function callback( $path = '', $blog_id = 0 ) {
 		}
 
 		$response = array();
+
+		// @start-hide-in-jetpack
```
Comment (jeherve) [inline json-endpoints/class.wpcom-json-api-list-media-v1-1-endpoint.php]: This seems like it made it in by mistake; that Image classification tool isn't part of Jetpack yet.

File: json-endpoints/class.wpcom-json-api-get-site-endpoint.php [hunk 2]
```diff
@@ -303,7 +334,7 @@ protected function render_option_keys( &$options_response_keys ) {
 		$site = $this->site;
 
 		$custom_front_page = $site->is_custom_front_page();
-
+
```
Comment (zinigor) [inline json-endpoints/class.wpcom-json-api-get-site-endpoint.php]: Added space.

File: json-endpoints/class.wpcom-json-api-get-site-endpoint.php [hunk 3]
```diff
@@ -368,7 +399,7 @@ protected function render_option_keys( &$options_response_keys ) {
 					$options[ $key ] = $site->get_image_large_width();
 					break;
 				case 'image_large_height' :
-					$options[ $key ] = $site->get_image_large_height();
+					$options[ $key ] = $site->get_image_large_height();
```
Comment (zinigor) [inline json-endpoints/class.wpcom-json-api-get-site-endpoint.php]: Added space.

File: json-endpoints/class.wpcom-json-api-site-settings-endpoint.php [hunk 4]
```diff
@@ -487,7 +508,7 @@ public function update_settings() {
 					// settings are stored as deletable numeric (all empty
 					// values as delete intent), validated as media image
 					if ( empty( $value ) || WPCOM_JSON_API::is_falsy( $value ) ) {
-						if ( delete_option( $key ) ) {
+						if ( delete_option( $key ) || apply_filters( 'rest_api_site_icon_cleared', false )  ) {
```
Comment (jeherve) [inline json-endpoints/class.wpcom-json-api-site-settings-endpoint.php]: We'll need a docblock here.

File: json-endpoints/class.wpcom-json-api-list-media-v1-1-endpoint.php (hunk 1 above)
Comment (zinigor) [inline json-endpoints/class.wpcom-json-api-list-media-v1-1-endpoint.php]: `hide-in-jetpack` didn't work here apparently.

File: json-endpoints/class.wpcom-json-api-upload-media-v1-1-endpoint.php [hunk 5]
```diff
@@ -16,32 +22,106 @@ function callback( $path = '', $blog_id = 0 ) {
...
+
 		if ( empty( $media_files ) && empty( $media_urls ) ) {
 			return new WP_Error( 'invalid_input', 'No media provided in input.' );
 		}
 
-		$create_media = $this->handle_media_creation_v1_1( $media_files, $media_urls, $media_attrs );
-		$media_ids = $create_media['media_ids'];
-		$errors = $create_media['errors'];
+		// For jetpack sites, we send the media via a different method, because the sync is very different.
+		$jetpack_sync = Jetpack_Media_Sync::summon( $blog_id );
```
Comment (zinigor) [inline json-endpoints/class.wpcom-json-api-upload-media-v1-1-endpoint.php]: That probably needs different handling depending on if it's `WPCOM` or not.

**Comment by zinigor** [comment]
Closing in favor of the project: https://github.com/Automattic/jetpack/projects/10

**Pull request closed without merging**